src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -210,6 +215,7 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     }
 
     if (!this._config.aspect_ratio) {
+      root.style.paddingBottom = "100%";
```

Comment:
Duplicate of <https://github.com/home-assistant/frontend/pull/5206/commits/3480f7037da7e79ff170ee4131df27337f771d20#diff-0720252cea84c217e8440c81823a92f2R227> now?

Reply:
I will rebase once they are merged

Reply:
I don't understand? Why are these changes to the logic of the map card needed?

Reply:
The aspect ratio was not optional before. So changing the return type of the stub config gave a type error because it needed aspect ratio. Given how this is supposed to be optional I updated the type which in return gave type errors in the card its self because now aspect ratio could possibly not be defined. aka the changes to the card

Reply:
Let's just change:
```ts
    const ratio = parseAspectRatio(this._config.aspect_ratio);
    const ar =
      ratio && ratio.w > 0 && ratio.h > 0
        ? `${((100 * ratio.h) / ratio.w).toFixed(2)}`
        : "100";
    return 1 + Math.floor(Number(ar) / 25) || 3;
  }
```
to:
```ts
    const ratio = this._config.aspect_ratio ? parseAspectRatio(this._config.aspect_ratio) : null;
    const ar =
      ratio && ratio.w > 0 && ratio.h > 0
        ? `${((100 * ratio.h) / ratio.w).toFixed(2)}`
        : "100";
    return 1 + Math.floor(Number(ar) / 25) || 3;
  }
```

Reply:
If we already know the calculation. Why go through all extra logic?

Reply:
Uuuh sorry, wrong code. I meant lines 222 and 224.

Reply:
No I know

I am saying why go through all the extra logic and calc. when if there is no ratio then its 5

Reply:
Ok, why do we return 3 when there is no config?

Reply:
/shrug

Reply:
This was how it was built haha. I think the comments below suggestion would be best right. Just change the `!this._config?.apsect_ratio` to 5